delivery_carrier_label_gls/models/delivery_carrier.py

```diff
@@ -0,0 +1,153 @@
+        selection=[
+            ("D_200", "D200"),
+            ("PF_4_I", "PF4I"),
+            ("PF_4_I_200", "PF4I200"),
+            ("PF_4_I_300", "PF4I300"),
+            ("PF_8_D_200", "PF8D200"),
+            ("T_200_BF", "T200BF"),
+            ("T_300_BF", "T300BF"),
+            ("ZPL_200", "ZPL200"),
+            ("ZPL_300", "ZPL300"),
+        ],
+        default=False,
+    )
+    gls_return_partner_id = fields.Many2one(
+        "res.partner",
+        string="Return Address",
+        help="If set, this partner's address will be used on the return label.",
+    )
+
+    @api.constrains("gls_return_partner_id")
+    def _check_gls_return_partner_id(self):
+        records_to_check = self.filtered(
+            lambda c: c.delivery_type == "gls" and c.gls_return_partner_id
+        )
+        for carrier in records_to_check:
+            carrier.gls_return_partner_id._gls_prepare_address()
+
+    @api.constrains(
+        "delivery_type",
+        "gls_contact_id",
+        "gls_url_tracking",
+        "gls_label_format",
+        "carrier_account_id",
+    )
+    def _check_gls_fields(self):
+        gls_field_names = [
+            "gls_contact_id",
+            "gls_url_tracking",
+            "gls_label_format",
+            "carrier_account_id",
+        ]
+        gls_records = self.filtered(lambda c: c.delivery_type == "gls")
+        for rec in gls_records:
+            for field_name in gls_field_names:
+                value = rec[field_name]
+                if not value:
+                    field = rec._fields[field_name]
+                    description_string = field._description_string(self.env)
+                    raise ValidationError(
+                        _(
+                            f"The GLS field '{description_string}' is required for "
+                            f"carrier {rec.name}"
+                        )
+                    )
+
+        for rec in self - gls_records:
+            if any(
+                rec[f]
+                for f in gls_field_names
+                if f not in ("carrier_account_id", "gls_label_format")
+            ):
+                raise ValidationError(
+                    _(f"Incorrect GLS parameters set on carrier {rec.name}.")
+                )
+
+    @api.constrains("delivery_type", "prod_environment", "gls_url", "gls_url_test")
+    def _check_gls_url(self):
+        for rec in self.filtered(lambda c: c.delivery_type == "gls"):
+            if not rec.prod_environment and not rec.gls_url_test:
+                raise ValidationError(
+                    _("The GLS field 'Test Service Url' is required in test mode")
+                )
+            if rec.prod_environment and not rec.gls_url:
+                raise ValidationError(
+                    _("The GLS field 'Service Url' is required in non test mode")
+                )
+
+    def gls_get_shipping_price_from_so(self, order):
+        self.ensure_one()
+        return [0]  # TODO?
```

Comment:
I can't find where this is used

Reply:
I removed it since it was probably a maybe more than a todo.